File: B2.java
package ejercicios;

import java.util.ArrayList;
import java.util.Scanner;
import java.util.regex.Pattern;

public class Ejercicios {

    public static void main(String[] args) {

        ArrayList<String> Lista = new ArrayList<>();
        Scanner in = new Scanner(System.in);

        boolean sw = true;

        while (sw) {

            String texto = in.nextLine();

            if (!texto.equals("-2")) {
                Lista.add(resp(texto));

            } else {
                sw = false;

            }

        }

        for (int i = 0; i < Lista.size(); i++) {
            System.out.println(Lista.get(i));
        }

    }

    public static String resp(String t) {
        String texto = t;

        String cantidad = texto.substring(texto.length() - 1, texto.length());
        String numeros = texto.substring(0, texto.length() - 5);

        String patron = "[ ]+";
        Pattern p1 = Pattern.compile(patron);
        String[] items = p1.split(numeros);
        String n[] = items;
        ArrayList<String> Lista = new ArrayList<>();

        for (String s : items) {
            Lista.add(s);
        }

        int Matriz[] = new int[Lista.size()];
        for (int i = 0; i < Matriz.length; i++) {

            Matriz[i] = Integer.parseInt(Lista.get(0));
            Lista.remove(0);

        }

        int aux = 0;
        String resp = "";
        for (int i = 0; i < Matriz.length; i++) {

            for (int j = i; j < Matriz.length; j++) {

                if ((j + 1) % Integer.parseInt(cantidad) == 0) {

                    for (int x = j; x >= i; x--) {

                        resp = resp + (Matriz[x]) + " ";
                    }

                    aux = j;
                    break;

                }

            }
            i = aux;

        }

        return resp;
    }

}

File: B2.py
def resp(t):
	texto = t
	
	cantidad = texto[len(texto) - 1 : len(texto)]
	numeros = texto[0 : len(texto) - 5]
	
	Matriz = numeros.split()
	
	aux = 0
	resp = ""

	aux = []
	Lista = []

	for i in range(len(Matriz)):
		aux.append(Matriz[i])
		if (i + 1) % int(cantidad) == 0:
			Lista.append(aux)
			aux = []

	
	resp = ""

	for i in Lista:
		i.reverse()
		for j in i :
			resp = resp + str(j) + " "
	
	return resp





				

sw = True
Lista = []


while sw:
	texto = input()

	if texto != "-2":
		Lista.append(resp(texto))
	else:
		sw = False


for i in Lista:
	print(i)
